### ISSUE TITLE
Rename Symbol on class resulting in file rename leaves 2 files if "Apply" while refactoring preview is displayed

### ISSUE THREAD

**Issue by still-dreaming-1** [DEVSENSE/phptools-docs#647]
**TLDR**: When using Rename Symbol on a class, if I click "Apply" while the refactoring preview is displayed, the original file still exists in addition to the renamed file.

**More detailed explanation**

If I use the Rename Symbol on a class, this will result in a checklist of possible refactorings. If I check the one to rename the file for that class, and click on any of the items in the checklist resulting in the refactoring preview being displayed, and then click the "Apply" button, it will create the renamed version of the file and rename references correctly, but it will also leave the original file in tact. The key here is that if the refactoring preview window is not displayed, the refactoring completes as desired, with only the renamed file in existence.

**Why it matters**

Due to my attempts to use a custom, keyboard only, as efficient as possible workflow, I often end up incidentally having the refactoring preview displayed when I "Apply" the refactorings, even though I don't really want to see the preview. This results in the described problem, slowing me down.

**Comment by jakubmisek** [comment]
@still-dreaming-1 this looks like some other process (maybe another extension?) is blocking the file from being renamed. I cannot repro the same behavior so far.

**Comment by still-dreaming-1** [comment]
You were right. I got it working better now. It was related to the vscode-neovim plugin. I should have at least tried reproducing it without my custom neovim key binding, which I was unable to. I got it working better with my binding by putting the key mapping functionality into an anonymous function like so:

```
vim.keymap.set(
  'n',
  '<leader>j',
  function()
    vscode.action('editor.action.rename')
  end
)
```

Whereas before when it was only mostly working right I had it like this:

```
vim.keymap.set(
  'n',
  '<leader>j',
  "<Cmd>lua require('vscode-neovim').action('editor.action.rename')<CR>"
)
```

**Comment by jakubmisek** [comment]
Thank you for the update. I'm glad you found the problem.

**Comment by still-dreaming-1** [comment]
Just an FYI. Actually this issue is still happening in my setup, but only intermittently. May not be anything you can/should do about it as it works fine as long as I use this feature as intended instead of launching the action through a vscode-neovim key binding.

**Comment by jakubmisek** [comment]
@still-dreaming-1 I'm sorry I can't provide better help as I'm unfamiliar with the setup above. 

Is it worth reporting the issue at `vscode-neovim` plugin?

**Comment by still-dreaming-1** [comment]
No problem. Yeah I would need to report it to them if I want to pursue it further. Thanks.
